javacord-api/src/main/java/org/javacord/api/DiscordApi.java

```diff
@@ -1436,7 +1436,7 @@ default CompletableFuture<Sticker> requestStickerById(String id) {
         try {
             return requestStickerById(Long.parseLong(id));
         } catch (NumberFormatException exception) {
-            throw new IllegalArgumentException("The ID must be a number!");
+            return requestStickerById(-1);
```

Comment:
```suggestion
            CompletableFuture<Sticker> future = new CompletableFuture<>();
            future.completeExceptionally(exception);
            return future;
```
There is no reason to swallow the exception and deprive the user of debugging information.

Reply:
Seems there is another inconsistency. `DiscordApi#getUserById` looks the same like this

Reply:
I'd argue we should return a failed future in both cases. Depending on how many methods have this problem we can create an issue or make the changes all in this PR. If we have a way to pass the exception to the user without breaking the method contract, we should do so and not throw away the information.